### PR TITLE
AP-10066 # Added hiddenFrom to _FormElementBase

### PR DESCRIPTION
**Pull request by mymattcarroll**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

---

> [!NOTE]
> **Low Risk**
> Public type and JSDoc changes only; backward compatible when `hiddenFrom` is omitted, though consumers must implement audience-aware hiding to use the new field.
> 
> **Overview**
> Extends form element typing so **hidden** fields can target specific audiences instead of always hiding from everyone.
> 
> Adds **`FormElementHiddenFromAudience`** (`FORM_COMPLETER`, `APPROVER`) and optional **`hiddenFrom`** on **`_FormElementBase`**. When **`isHidden`** is true and **`hiddenFrom`** is omitted, behaviour stays the same (hidden from completers and approvers). When **`hiddenFrom`** is set, the element is hidden only from those audiences. **`isHidden`** docs are updated to describe this interaction.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 400cdcf1b17e83386726c2c21a2b51f9ab597c57. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: typescript/forms.d.ts [hunk 1]
```diff
@@ -41,6 +41,14 @@ import { ArcGISWebMapElementValue } from './arcgis'
 
 export type HintPositionEnum = 'BELOW_LABEL' | 'TOOLTIP'
 
+/**
+ * Audiences that a hidden form element can be restricted to.
+ *
+ * - `FORM_COMPLETER` — users completing the form
+ * - `APPROVER` — approvers reviewing the form submission
+ */
+export type FormElementHiddenFromAudience = 'FORM_COMPLETER' | 'APPROVER'
```
Comment (kizaonline) [inline typescript/forms.d.ts:50]: Is an option for hiding from both valid?

Comment (kizaonline) [reply typescript/forms.d.ts:50]: AH isHidden would do that

**Review by kizaonline — APPROVED** [review]

File: typescript/forms.d.ts (hunk 1 above)
Comment (mymattcarroll) [reply typescript/forms.d.ts:50]: Correcto

Comment (mymattcarroll) [reply typescript/forms.d.ts:50]: (Thats Spanish for correct)

**Review by divporter — APPROVED** [review]
